core/src/test/java/org/apache/iceberg/deletes/TestEqualityFilter.java

```diff
@@ -63,14 +64,17 @@ public void testEqualitySetFilterLongColumn() {
             Row.of(7L, "g", "grizzly"),
             Row.of(8L, "h", null));
 
-    Assert.assertEquals(
-        "Filter should produce expected rows",
-        expected,
+    List<StructLike> actual =
         Lists.newArrayList(
             Deletes.filter(
                 ROWS,
                 row -> Row.of(row.get(0, Long.class)),
-                Deletes.toEqualitySet(deletes, ROW_SCHEMA.select("id").asStruct()))));
+                Deletes.toEqualitySet(deletes, ROW_SCHEMA.select("id").asStruct())));
+
+    assertThat(actual)
+        .as("Filter should produce expected rows")
+        .usingRecursiveComparison()
```

Comment:
why do we need this one? The test passes for me without having `usingRecursiveComparison()`